contrib_bots/lib/commute_bot.py

```diff
@@ -0,0 +1,263 @@
+            response in the same stream or through private messages if the
+            input was originally private.
+
+            Sample input:
+            @commute origins=Chicago,IL,USA destinations=New+York,NY,USA
+            @commute help
+            '''
+
+    help_info = '''
+    Obligatory Inputs:
+        Origin e.g. origins=New+York,NY,USA
+        Destination e.g. destinations=Chicago,IL,USA
+    Optional Inputs:
+        Mode Of Transport (inputs: driving, walking, bicycling, transit)
+        Default mode (no mode input) is driving
+        e.g. mode=driving or mode=transit
+        Units (metric or imperial)
+        e.g. units=metric
+        Restrictions (inputs: tolls, highways, ferries, indoor)
+        e.g. avoid=tolls
+        Departure Time (inputs: now or (YYYY, MM, DD, HH, MM, SS) departing)
+        e.g. departure_time=now or departure_time=2016,12,17,23,40,40
+        Arrival Time (inputs: (YYYY, MM, DD, HH, MM, SS) arriving)
+        e.g. arrival_time=2016,12,17,23,40,40
+        Languages:
+        Languages list: https://developers.google.com/maps/faq#languagesupport)
+        e.g. language=fr
+
+    Sample request:
+        @commute origins=Chicago,IL,USA destinations=New+York,NY,USA language=fr
+
+    Please note:
+        Fare information can be derived, though is solely dependent on the
+        availability of the informatipython run.py lib/followup.py --config-file ~/.zuliprc-localon released by public transport operators.
+        Duration in traffic can only be derived if a departure time is set.
+        If a location has spaces in its name, please use a + symbol in the
+        place of the space/s.
+        A departure time and a arrival time can not be inputted at the same time
+        To add more than 1 input for a category,
+        e.g. more than 1 destinations,
+        use (|), e.g. destinations=Empire+State+Building|Statue+Of+Liberty
+        No spaces within addresses.
+        Departure times and arrival times must be in the UTC timezone,
+        you can use the timezone bot.
+                '''
+
+    def triage_message(self, message):
+        # return True iff we want to (possibly) response to this message
+        original_content = message['content']
+        # This next line of code is defensive, as we
+        # never want to get into an infinite loop of posting follow
+        # ups for own follow ups!
+        if message['display_recipient'] == 'commute':
+            return False
+        is_commute = original_content.startswith('@commute')
+        return is_commute
+
+    # adds API Authentication Key to url request
+    def get_api_key(self):
+        # google-commute.ini must have been moved from
+        # ~/zulip/contrib_bots/lib/CommuteBot/google-commute.ini into
+        # /google-commute.ini
+        # see doc.md for more information
+        home = expanduser('~')
+        settings = open(home + '/google-commute.ini')
```

Comment:
And it would be better to define a global variable for the path like this `CONFIG_PATH = home + '/google-commute.ini'`.